File: frontend/src/services/consumableService.js
import axios from '../api/axios';

const getAllConsumables = async () => {
    const response = await axios.get('/consumables');
    return response.data;
};

const createConsumable = async (consumable) => {
    const response = await axios.post('/consumables', consumable);
    return response.data;
};

const updateConsumable = async (id, consumable) => {
    const response = await axios.put(`/consumables/${id}`, consumable);
    return response.data;
};

const deleteConsumable = async (id) => {
    await axios.delete(`/consumables/${id}`);
};

export default {
    getAllConsumables,
    createConsumable,
    updateConsumable,
    deleteConsumable
};

File: mobile/lib/data/services/consumable_service.dart
import 'package:dio/dio.dart';
import '../../core/constants/api_constants.dart';
import 'api_service.dart';

class ConsumableService {
  final ApiService _apiService = ApiService();

  Future<List<dynamic>> getConsumables() async {
    try {
      final response = await _apiService.dio.get(ApiConstants.consumablesEndpoint);
      if (response.statusCode == 200 && response.data != null) {
        return response.data as List<dynamic>;
      }
      return [];
    } catch (e) {
      print('Consumable Service Error: $e');
      throw e;
    }
  }

  Future<void> deleteConsumable(int id) async {
    try {
      await _apiService.dio.delete('${ApiConstants.consumablesEndpoint}/$id');
    } catch (e) {
      print('Error deleting consumable: $e');
      rethrow;
    }
  }

  Future<void> createConsumable(Map<String, dynamic> data) async {
    try {
      await _apiService.dio.post(ApiConstants.consumablesEndpoint, data: data);
    } catch (e) {
      print('Error creating consumable: $e');
      rethrow;
    }
  }

  Future<void> updateConsumable(int id, Map<String, dynamic> data) async {
    try {
      await _apiService.dio.put('${ApiConstants.consumablesEndpoint}/$id', data: data);
    } catch (e) {
      print('Error updating consumable: $e');
      rethrow;
    }
  }
}
